### PR TITLE
[feat] Aim integration

### PR DESCRIPTION
**Pull request by tamohannes**
### Background
Integrating with Aim would be fantastic, as it would provide a convenient method for tracking metadata and storing experiment histories in an intuitive and visually pleasing manner.
To enhance experiment reproducibility and provide a unified, aesthetically pleasing user interface for exploring experiments and their corresponding input/outputs, it is desirable to furnish users with a fast and a solid tool. Aim facilitates gaining deeper insights, thus enhancing the quality of experimentation.

### Changes
The following changes are made in the codebase
- Created `AimCallback` to track input prompts and agent replies, in a separate `aim.py` file.
- Added cli option `--aim-repo` to give a specific Aim repo path if needed.
- Passed an instance of `AimCallback`: `aim_callback` to track the logs in `start_interaction_loop` method 

### Documentation
I've included in-code comments to make it easy for the reviewer to understand.

### Test Plan
I executed Auto-GPT with some tasks and thoroughly explored the Aim UI during the response period of the interaction loop.

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes 

There is no need for any specific tests. By running Auto-GPT and examining Aim explorer, you will have all the information you need.

### CODE REVIEW

**Comment by ntindle** [comment]
Hey! Would love some more context about aim in the PR and why it’s useful

**Comment by tmynn** [comment]
Hello @ntindle, thank you for your prompt response. Aim is an open-source AI metadata tracker that is designed to be user-friendly and highly efficient. With Aim, you can effortlessly log all your AI metadata, and utilize its intuitive UI to observe and compare them. Additionally, Aim comes equipped with an SDK that enables you to query metadata programmatically.

By using Aim, you can ensure the reproducibility of your experiments, as it helps you to keep track of all the generated responses. Overall, Aim is a powerful tool that simplifies AI metadata management and enhances the efficiency of your workflow.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

File: .github/workflows/pr-label.yml [hunk 1]
```diff
@@ -26,23 +26,3 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          s_label: "size/s"
-          s_max_size: "10"
-          m_label: "size/m"
-          m_max_size: "50"
-          l_label: "size/l"
-          l_max_size: "200"
-          xl_label: "size/xl"
-          fail_if_xl: "false"
-          github_api_url: "api.github.com"
```
Comment (anonhostpi) [inline .github/workflows/pr-label.yml]: Why would you delete this?

This is for Github Continuous Integration.

**Review by anonhostpi — CHANGES_REQUESTED** [review]
Why are you removing a github actions bot?

**Comment by ntindle** [comment]
@merwanehamadi

**Review by ntindle — CHANGES_REQUESTED** [review]
Fix changes to unrelated files

**Comment by tmynn** [comment]
Hey @ntindle, it seems like there are numerous unrelated files present, which is making the repository look messy. Let me open another PR to keep things clean.
By the way, I wanted to check if you've completed the phase of re-architecting the core. Is it relatively stable now, so I can proceed with opening the PR?

**Comment by tmynn** [comment]
@ntindle, nevermind. don't need to open a new PR. have successfully resolved the conflicts. :)

**Comment by ntindle** [comment]
Ref: https://github.com/aimhubio/aim

**Comment by tmynn** [comment]
Resolves #4842

**Pull request closed without merging**